pkg/network/usm/ebpf_main.go

```diff
@@ -273,13 +273,20 @@ func (e *ebpfProgram) Start() error {
 // Close stops the ebpf program and cleans up all resources.
 func (e *ebpfProgram) Close() error {
 	e.mapCleaner.Stop()
+	ebpftelemetry.UnregisterTelemetry(e.Manager.Manager)
+	var err error
+	for _, pm := range e.PerfMaps {
+		err = errors.Join(err, pm.Stop(manager.CleanAll))
+	}
+	for _, rb := range e.RingBuffers {
+		err = errors.Join(err, rb.Stop(manager.CleanAll))
+	}
```

Comment:
Perhaps it would be useful to add a comment explaining why we need to do stop the PerfMaps and RingBuffers here despite it also being done in e.Stop().